src/openloco/windows/construction/ConstructionTab.cpp

```diff
@@ -0,0 +1,1276 @@
+            {
+                auto xPos = _mapSelectedTiles[i].x;
+                while (xPos != -1)
+                {
+                    auto yPos = _mapSelectedTiles[i].y;
+
+                    if (xPos >= 0x2FFF)
+                    {
+                        i++;
+                        xPos = _mapSelectedTiles[i].x;
+                        continue;
+                    }
+
+                    if (yPos >= 0x2FFF)
+                    {
+                        i++;
+                        xPos = _mapSelectedTiles[i].x;
+                        continue;
+                    }
+
+                    getConstructionHeight(_mapSelectedTiles[i], &roadHeight, true);
+
+                    i++;
+                    xPos = _mapSelectedTiles[i].x;
+                }
+            }
+            // loc_4A23F8
+            _word_1136000 = roadHeight;
+            _mapSelectionFlags = _mapSelectionFlags & ~(0x7);
+
+            registers regs2;
+            regs2.ax = x;
+            regs2.bx = y;
+            auto flags = call(0x00478361, regs2);
+            map_pos mapPos = { regs2.ax, regs2.bx };
+
+            if (!(flags & 1 << 8))
+            {
+                auto pos = screenGetMapXyWithZ(xy32(x, y), roadHeight << 3);
+                if (pos)
+                {
+                    mapPos.x = (*pos).x;
+                    mapPos.y = (*pos).y;
+                    mapPos.x &= 0xFFE0;
+                    mapPos.y &= 0xFFE0;
+                    _byte_113605D = 1;
+                    _word_1135FFE = roadHeight;
+                }
+                else
+                {
+                    mapPos.x = -1;
+                }
+            }
+
+            if (flags & 1 << 8 || mapPos.x == -1)
+            {
+                registers regs3;
+                regs3.ax = x;
+                regs3.bx = y;
+                regs3.edi = _mapSelectedTiles[i].x << 16 | _mapSelectedTiles[i].x;
+                call(0x00460781, regs3);
+
+                if (regs3.ax == -32768)
+                    return;
+
+                mapPos.x = regs3.ax;
+                mapPos.y = regs3.bx;
+
+                getConstructionHeight(mapPos, &roadHeight, true);
+
+                _byte_113605D = 0;
+            }
+            input::cancel_tool();
+
+            auto ebx = 0;
+            if (input::has_key_modifier(input::key_modifier::shift) || _byte_113605D != 1)
+            {
+                auto roadPiece = common::roadPieces[_byte_1136065];
+                i = 0;
+                auto height = 0;
+
+                while (roadPiece[i].index != 0xFF)
+                {
+                    if (height > roadPiece[i].z)
+                        height = roadPiece[i].z;
+                    i++;
+                }
+
+                roadHeight -= height;
+                roadHeight -= 16;
+                ebx = 2;
+
+                if (input::has_key_modifier(input::key_modifier::shift))
+                {
+                    ebx = 0x80000000;
```

Comment:
0x80000008

This is the number of retries that it will do when holding down shift.